src/critnib.h

```diff
@@ -30,13 +30,24 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#ifndef CRITNIB_H
+#define CRITNIB_H
+
 #include "vmemcache.h"
+#include "os_thread.h"
+
+struct critnib_node;
+struct critnib {
+	struct critnib_node *root;
+	os_mutex_t lock_index;
```

Comment:
I don't like the fact that we are breaking abstraction. Maybe we should create the `vmemcache_index` structure which would contain the critnib and the lock?

Reply:
We already have — at least functionally (an empty index is just a NULL).  But perhaps this is what you're proposing (ie, a naming change)?

Note that I'm about to break encapsulation the other direction, by using knowledge of cache_entry inside critnib: every leaf is a pointless two-pointer struct where one pointer is at a fixed offset from the other.  Replacing the leaf with cache_entry would avoid all those allocs at no apparent cost.